examples/timestamping/backend/tests/config.rs

```diff
@@ -0,0 +1,137 @@
+
+use exonum_merkledb::access::AccessExt;
+use exonum_rust_runtime::ServiceFactory;
+use exonum_supervisor::{ConfigPropose, Supervisor};
+use exonum_testkit::{TestKit, TestKitBuilder};
+use exonum_time::{MockTimeProvider, TimeServiceFactory};
+
+use std::time::SystemTime;
+
+use exonum_timestamping::{Config, TimestampingService};
+
+const TIME_SERVICE_ID: InstanceId = 102;
+const TIME_SERVICE_NAME: &str = "time";
+const SERVICE_ID: InstanceId = 103;
+const SERVICE_NAME: &str = "timestamping";
+const SECOND_TIME_SERVICE_ID: InstanceId = 104;
+const SECOND_TIME_SERVICE_NAME: &str = "time2";
+
+fn init_testkit(second_time_service: bool) -> (TestKit, MockTimeProvider) {
+    let mock_provider = MockTimeProvider::new(SystemTime::now().into());
+    let time_service = TimeServiceFactory::with_provider(mock_provider.clone());
+    let time_service_artifact = time_service.artifact_id();
+    let timestamping = TimestampingService;
+    let timestamping_artifact = timestamping.artifact_id();
+
+    let testkit = TestKitBuilder::validator()
+        .with_rust_service(Supervisor)
+        .with_artifact(Supervisor.artifact_id())
+        .with_instance(Supervisor::simple())
+        .with_rust_service(time_service)
+        .with_rust_service(timestamping)
+        .with_artifact(time_service_artifact.clone())
+        .with_instance(
+            time_service_artifact.into_default_instance(TIME_SERVICE_ID, TIME_SERVICE_NAME),
+        )
+        .with_artifact(timestamping_artifact.clone())
+        .with_instance(
+            timestamping_artifact
+                .into_default_instance(SERVICE_ID, SERVICE_NAME)
+                .with_constructor(Config {
+                    time_service_name: TIME_SERVICE_NAME.to_owned(),
+                }),
+        );
+
+    let testkit = if second_time_service {
+        let time_service = TimeServiceFactory::with_provider(mock_provider.clone());
+        let time_service_artifact = time_service.artifact_id();
+        testkit
+            .with_rust_service(time_service)
+            .with_artifact(time_service_artifact.clone())
+            .with_instance(
+                time_service_artifact
+                    .into_default_instance(SECOND_TIME_SERVICE_ID, SECOND_TIME_SERVICE_NAME),
+            )
+            .build()
+    } else {
+        testkit.build()
+    };
+
+    (testkit, mock_provider)
+}
+
+/// Creates block with `ConfigPropose` tx and returns current service configuration.
+fn propose_configuration(testkit: &mut TestKit, config: Config) -> Config {
+    let tx = ConfigPropose::immediate(0).service_config(SERVICE_ID, config);
+
+    let initiator_id = testkit.network().us().validator_id().unwrap();
+    let (pub_key, sec_key) = &testkit.validator(initiator_id).service_keypair();
+    testkit.create_block_with_transaction(tx.sign_for_supervisor(*pub_key, sec_key));
+
+    testkit
+        .snapshot()
+        .for_service(SERVICE_NAME)
+        .unwrap()
+        .get_entry("config")
```

Comment:
Nit: This looks fragile. I'd rather suggest to use the HTTP API of the service to retrieve the config. Also, to test configuration failures, it could be more appropriate to check the error of the corresponding proposal.